### PR TITLE
test(walrs_validation): #144 Add NaN and infinity assertion tests for float rules

### LINKED ISSUE #144: test(walrs_validation): Add NaN assertion tests for float validation rules
## Summary

Existing NaN edge-case tests only `println!` results — they never `assert!`. This means the NaN validation bug is not caught by CI.

## What to Do

- Add assertion-based tests for `f32`/`f64` with `NAN`, `INFINITY`, and `NEG_INFINITY` against `Min`, `Max`, `Range`, `Step`, `Equals`, and `OneOf` rules.
- Verify expected behavior and lock it down with assertions.

## Files

- Tests in `crates/validation/src/rule_impls/steppable.rs` (inline `#[cfg(test)]` module)
- Tests in `crates/validation/src/rule_impls/scalar.rs` (inline `#[cfg(test)]` module)

## Priority

High — directly related to the NaN validation bug.

### PR DESCRIPTION
**Pull request by elycruz**
## Summary

Closes #144.

### NaN tests for missing rules
- `Step`, `Equals`, `OneOf` rules with `f32::NAN` / `f64::NAN` in both **steppable.rs** and **scalar.rs**
- Verifies all return `Err` (except scalar `Step` which is a pass-through → `Ok`)

### INFINITY / NEG_INFINITY tests
- All 6 rules (`Min`, `Max`, `Range`, `Step`, `Equals`, `OneOf`) tested with `f32`/`f64` infinity variants
- Verified actual behavior and locked down with assertions:
  - `Min(0.0)`: INFINITY passes, NEG_INFINITY fails
  - `Max(100.0)`: INFINITY fails, NEG_INFINITY passes
  - `Range`: INFINITY/NEG_INFINITY both fail
  - `Step`: steppable returns Err (NaN remainder), scalar passes through
  - `Equals`/`OneOf`: INFINITY/NEG_INFINITY both fail

**60 new assertion-based tests** added across steppable.rs and scalar.rs. All tests passing.

### CODE REVIEW

File: crates/validation/src/violation.rs [hunk 1]
```diff
@@ -74,7 +77,10 @@ impl Violation {
   pub fn exact_length(expected: usize, actual: usize) -> Self {
     Self::new(
       ViolationType::TooShort,
-      format!("Value length must be exactly {} (got {}).", expected, actual),
+      format!(
+        "Value length must be exactly {} (got {}).",
+        expected, actual
+      ),
     )
```
[the remark this answers is not in the document]
Comment (elycruz) [reply crates/validation/src/violation.rs:84]: No, for now this implementation is OK.

File: crates/validation/src/rule_impls/steppable.rs [hunk 2]
```diff
@@ -718,6 +718,255 @@ mod tests {
     assert!(rule.validate_ref(&Some(-0.1)).is_err());
   }
 
+  // ========================================================================
+  // NaN Validation Tests — Step, Equals, OneOf (f32 / f64)
+  // ========================================================================
+
```
[the remark this answers is not in the document]
Comment (elycruz) [reply crates/validation/src/rule_impls/steppable.rs:724]: This is OK, since code wasn't previously formatted.

**Comment by elycruz** [comment]
@claude[agent] Please resolve conflicts in this [PR]( #179 ).
